### DISCUSSION TITLE
Measurable accessibility and quality workflow

### DISCUSSION THREAD

**Discussion by soykje** [adevinta/spark#797] [Ideas]
Hi everyone! :wave:

As Spark is trying to provide a strong Design System with accessibility "out of the box", we also want to install a strong and reliable workflow to develop & test our components, on an accessibility basis specifically.

## Tools
There are a few tools we should agree to use, I guess:
- [aXe Chrome extension](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd)
- Lighthouse Chrome extension
- [aXe React package](https://github.com/dequelabs/axe-core-npm/tree/develop/packages/react) which replaces the deprecated React a11y one.
- [aXe StoryBook extension](https://storybook.js.org/addons/@storybook/addon-a11y) which is also documented [here](https://storybook.js.org/docs/react/writing-tests/accessibility-testing).

## Documentation
A documented guideline may also be helpful in our SB (just as the one from [Radix](https://www.radix-ui.com/docs/primitives/overview/accessibility)). We could write about our tooling and our quality workflow (tests, CI, ...), and how each contributor could be part of it :+1: 

## CI
I think we also should try to define the accessibility level we want to be compliant with (WCAG A, AA, AAA...), and maybe setup a CI check task that could help us improve our work on accessible features?

**Comment by soykje** [comment]
@andresin87 suggested https://playwright.dev/docs/accessibility-testing... Maybe we should also have a look :+1:

**Comment by soykje** [comment]
@Powerplex just proposed to [add the StoryBook a11y from Axe plugin](https://github.com/adevinta/spark/pull/834). We may after be able to add some a11y checks to CI. :muscle:

**Reply by soykje** [reply]
After this addition, we worked on a11y check on CI: https://github.com/adevinta/spark/issues/878

**Comment by soykje** [comment]
Should we add a dedicated section at the root of SB/Zeroheight? Would be about explaining our a11y interest and concerns, the WCAG level we took as reference, and few other things... cc @Powerplex @turolopezsanabria

**Comment by Powerplex** [comment]
`axe-core` setup is complete and working. Closing.
